specification/cosmos-db/resource-manager/Microsoft.DocumentDB/preview/2021-04-01-preview/services.json

```diff
@@ -0,0 +1,399 @@
+          "DataTransfer",
+          "SqlDedicatedGateway"
+        ],
+        "x-ms-examples": {
+          "DataTransferServiceDelete": {
+            "$ref": "./examples/CosmosDBDataTransferServiceDelete.json"
+          },
+          "SqlDedicatedGatewayServiceDelete": {
+            "$ref": "./examples/CosmosDBSqlDedicatedGatewayServiceDelete.json"
+          }
+        },
+        "operationId": "Service_Delete",
+        "description": "Deletes service with the given serviceName.",
+        "x-ms-long-running-operation": true,
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/accountNameParameter"
+          },
+          {
+            "$ref": "#/parameters/serviceNameParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "202": {
+            "description": "The job will complete asynchronously."
+          },
+          "204": {
+            "description": "The response for service deletion."
+          },
+          "200": {
+            "description": "The response for service deletion."
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "ServicesCreateUpdateParameters": {
+      "description": "Service create or update parameters.",
+      "type": "object",
+      "properties": {
+        "properties": {
+          "type": "object",
+          "x-ms-client-flatten": true,
+          "description": "The properties for creating services.",
+          "$ref": "#/definitions/ServicesCreateUpdateProperties"
+        }
+      },
+      "required": [
+        "properties"
+      ],
+      "x-ms-azure-resource": true
+    },
+    "ServicesCreateUpdateProperties": {
+      "description": "Describes the service response property.",
+      "type": "object",
+      "properties": {
+        "instanceSize": {
+          "$ref": "#/definitions/ServiceResourceInstanceSize"
+        },
+        "instanceCount": {
+          "description": "Instance count for the service.",
+          "type": "integer"
+        },
+        "serviceType": {
+          "type": "string",
+          "description": "The type of Azure resource."
+        }
+      },
+      "required": [
+        "instanceSize",
+        "instanceCount"
+      ]
+    },
+    "ServiceResource": {
+      "description": "Services response resource.",
+      "type": "object",
+      "discriminator": "serviceType",
+      "allOf": [
+        {
+          "$ref": ".../../cosmos-db.json#/definitions/ARMProxyResource"
+        }
+      ],
+      "properties": {
+        "creationTime": {
```

Comment:
>creationTime [](http://example.com/codeflow?start=9&length=12)

Custom properties should be in the `properties` property and not at the top-level. This was also flagged by the linter.

Reply:
Fixed. Although I have never understood the reason for doing so. Given we use x-ms-client-flatten always.

Reply:
It's a convention that was established to have a separation between ARM-owned properties and RP-owned properties. It allows ARM to have consistent experience in the top-level properties (for example `identity` or `tags`) but still allow extensibility.